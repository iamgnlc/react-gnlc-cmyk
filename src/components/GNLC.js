import React from "react"

import g from "../assets/images/g.svg"
import n from "../assets/images/n.svg"
import l from "../assets/images/l.svg"
import c from "../assets/images/c.svg"

const letters = [
  { key: "g", img: g },
  { key: "n", img: n },
  { key: "l", img: l },
  { key: "c", img: c },
]

const GNLC = React.memo(() => {
  return letters.map((letter, i) => {
    return (
      <div key={i} className={`animated fadeInRight letter ${letter.key}`}>
        <img
          src={letter.img}
          alt={letter.key}
          className="animated fadeInLeft"
        />
      </div>
    )
  })
})

export default GNLC
