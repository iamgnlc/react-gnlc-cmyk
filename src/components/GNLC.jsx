import React, { memo } from 'react';

import g from '../assets/images/g.svg';
import n from '../assets/images/n.svg';
import l from '../assets/images/l.svg';
import c from '../assets/images/c.svg';

const letters = [
  { key: 'g', img: g },
  { key: 'n', img: n },
  { key: 'l', img: l },
  { key: 'c', img: c },
];

const ANIMATED_PREFIX = 'animate__';
const ANIMATED_MAIN_CLASS = `${ANIMATED_PREFIX}animated`;

const Letter = memo(({ letter }) => (
  <img
    src={letter.img}
    alt={letter.key}
    className={`${ANIMATED_MAIN_CLASS} ${ANIMATED_PREFIX}fadeInLeft`}
  />
));

Letter.displayName = 'Letter';

const GNLC = React.memo(() => {
  return letters.map((letter, i) => {
    return (
      <div
        key={`${letter.key}-${i}`}
        className={`${ANIMATED_MAIN_CLASS} ${ANIMATED_PREFIX}fadeInRight letter ${letter.key}`}
      >
        <Letter letter={letter} />
      </div>
    );
  });
});

GNLC.displayName = 'GNLC';

export default GNLC;
