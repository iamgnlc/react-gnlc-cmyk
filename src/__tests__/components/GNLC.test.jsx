import React from 'react';
import { render } from '@testing-library/react';

import GNLC from '../../components/GNLC';

describe('GNLC', () => {
  it('should render correctly', () => {
    const view = render(<GNLC />);

    expect(view.container).toMatchSnapshot();
  });
});
