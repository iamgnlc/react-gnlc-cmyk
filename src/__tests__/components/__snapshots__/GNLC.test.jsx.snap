// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GNLC should render correctly 1`] = `
<div>
  <div
    class="letter g"
  >
    <img
      alt="g"
      src="g.svg"
    />
  </div>
  <div
    class="letter n"
  >
    <img
      alt="n"
      src="n.svg"
    />
  </div>
  <div
    class="letter l"
  >
    <img
      alt="l"
      src="l.svg"
    />
  </div>
  <div
    class="letter c"
  >
    <img
      alt="c"
      src="c.svg"
    />
  </div>
</div>
`;
